spa/src/components/molecules/InfoBox.vue

```diff
@@ -0,0 +1,129 @@
+<template>
+  <div class="info-box"  v-if="infoBoxContent">
+    <span class="closebtn" @click="ignoreMessage()"><BIconXCircle scale='1'></BIconXCircle></span>
+    <component v-if="infoBoxContent" :is="infoBoxContent"></component>
+  </div>
+</template>
+
+<script>
+import { BIconXCircle } from 'bootstrap-vue';
+import { infoBoxes } from '../../../config/infoBox.json';
+
+export default {
+  name: 'InfoBox',
+  components: {
+    BIconXCircle,
+  },
+  data() {
+    return {
+      infoBoxContent: null,
+      infoBoxId: null,
+    };
+  },
+  mounted() {
+    this.loadInfoBox();
+  },
+  methods: {
+    loadInfoBox() {
+      const userIgnored = this.checkIfIgnored();
+      if (userIgnored) {
+        return;
+      }
+      const validInfoBoxes = this.findValidInfoBoxes(infoBoxes);
+      const randomIndex = this.randomNumber(validInfoBoxes.length);
```

Comment:
what if there is no valid info box? Length attribute will produce an error

Reply:
It would always return an array. But can avoid redundant evaluation of code if there is a check for array length. So changed.